crates/bevy_input/src/gamepad.rs

```diff
@@ -1,12 +1,20 @@
 use crate::{Axis, Input};
 use bevy_app::{EventReader, EventWriter};
 use bevy_ecs::system::{Res, ResMut};
-use bevy_utils::HashMap;
+use bevy_utils::{tracing::info, HashMap, HashSet};
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
 pub struct Gamepad(pub usize);
 
+#[derive(Default)]
+/// Container of unique connected [Gamepad]s
+///
+/// [Gamepad]s are registered and deregistered in [gamepad_connection_system]
+pub struct GamepadLobby {
+    pub gamepads: HashSet<Gamepad>,
```

Comment:
This is an implementation detail that we shouldn't expose directly to users (who could accidentally remove "active" gamepad and break guarantees that this list contains "active" gamepads). Instead, lets make this private and implement a method like `GamepadLobby::contains(index: usize)`

Reply:
Eh actually we should probably accept the `Gamepad` type for consistency with other apis, but I think we should consider making the index private to encourage people to treat `Gamepad` as an opaque type that must be queried for. And if we don't want that behavior (and want user-specified ids to be a part of our api), then we should consider removing the `Gamepad` wrapper type.

Also, lets add `GamepadLobby::iter()` so users can iterate active gamepads.